panel/io/handlers.py

```diff
@@ -180,11 +180,182 @@ def capture_code_cell(cell):
+                post_check()
+        except Exception as e:
+            autoreload_handle_exception(self, module, e)
+        finally:
+            # undo sys.path, CWD fixups
+            os.chdir(_cwd)
+            sys.path = _sys_path
+            sys.argv = _sys_argv
+            self.ran = True
+
+
+class PanelCodeHandler(CodeHandler):
+    """Modify Bokeh documents by creating Dashboard from code.
+
+    Additionally this subclass adds support for the ability to:
+
+    - Log session launch, load and destruction
+    - Capture document_ready events to track when app is loaded.
+    - Add profiling support
+    - Ensure that state.curdoc is set
+    - Reload the application module if autoreload is enabled
+    - Track modules loaded during app execution to enable autoreloading
+    """
+
+    def __init__(self, *, source: str, filename: PathLike, argv: list[str] = [], package: ModuleType | None = None) -> None:
+        Handler.__init__(self)
+
+        self._runner = PanelCodeRunner(source, filename, argv, package=package)
+
+        self._loggers = {}
+        for f in PanelCodeHandler._io_functions:
+            self._loggers[f] = self._make_io_logger(f)
+
+    def modify_document(self, doc: 'Document'):
+        if config.autoreload:
+            path = self._runner.path
+            argv = self._runner._argv
+            handler = type(self)(filename=path, argv=argv)
+            self._runner = handler._runner
+
+        module = self._runner.new_module()
+
+        # If no module was returned it means the code runner has some permanent
+        # unfixable problem, e.g. the configured source code has a syntax error
+        if module is None:
+            return
+
+        # One reason modules are stored is to prevent the module from being gc'd
+        # before the document is. A symptom of a gc'd module is that its globals
+        # become None. Additionally stored modules are used to provide correct
+        # paths to custom models resolver.
+        doc.modules.add(module)
+
+        run_app(self, module, doc)
 
-class MarkdownHandler(CodeHandler):
-    ''' Modify Bokeh documents by creating Dashboard from a Markdown file.
+CodeHandler.modify_document = PanelCodeHandler.modify_document
 
-    '''
+
+class ScriptHandler(PanelCodeHandler):
+    """Modify Bokeh documents by creating Dashboard from a Python script.
+    """
+
+
+    _logger_text = "%s: call to %s() ignored when running scripts with the 'bokeh' command."
+
+    _origin = "Script"
+
+    def __init__(self, *, filename: PathLike, argv: list[str] = [], package: ModuleType | None = None) -> None:
+        '''
+
+        Keywords:
+            filename (str) : a path to a Python source (".py") file
+
+        '''
+        with open(filename, encoding='utf-8') as f:
+            source = f.read()
+
+        super().__init__(source=source, filename=filename, argv=argv, package=package)
+
+bokeh.application.handlers.directory.NotebookHandler = ScriptHandler
```

Comment:
```suggestion
bokeh.application.handlers.directory.ScriptHandler = ScriptHandler
```
It's overwritten further down.